pkg/services/ngalert/models/alert_rule.go

```diff
@@ -186,6 +187,31 @@ func (alertRule *AlertRule) Diff(rule *AlertRule, ignore ...string) cmputil.Diff
 	return reporter.Diffs
 }
 
+// SetDashboardAndPanel will set the DashboardUID and PanlID
+// field be doing a lookup in the annotations. Errors when
+// the found annotations are not valid.
+func (alertRule *AlertRule) SetDashboardAndPanel() error {
+	if alertRule.Annotations == nil {
+		return nil
+	}
+	dashUID := alertRule.Annotations[DashboardUIDAnnotation]
+	panelID := alertRule.Annotations[PanelIDAnnotation]
+	if dashUID != "" && panelID == "" || dashUID == "" && panelID != "" {
+		return fmt.Errorf("both annotations %s and %s must be specified",
+			DashboardUIDAnnotation, PanelIDAnnotation)
+	}
+	if dashUID != "" {
```

Comment:
and then we can remove this if statement. I think it would make the code easier to read?